R/layer_residual_quantiles.R

```diff
@@ -68,12 +73,17 @@ slather.layer_residual_quantiles <-
     if (is.null(object$probs)) return(components)
 
     s <- ifelse(object$symmetrize, -1, NA)
-    r <- grab_residuals(the_fit, components)
-    q <- quantile(c(r, s * r), probs = object$probs, na.rm = TRUE)
+    r <- dplyr::bind_cols(
+      r = grab_residuals(the_fit, components),
+      geo_value = components$mold$extras$roles$geo_value,
+      components$mold$extras$roles$key) %>%
+      dplyr::group_by(dplyr::across(-r)) %>%
```

Comment:
Maybe I'm missing something in the code but does this grouping need to evaluate whether `object$by_key` is `TRUE` or `FALSE` before performing the group by?

Reply:
No you're not missing anything. Current implementation ignores that option and assumes it's TRUE. I haven't adjusted it to handle the FALSE case yet.

Reply:
I've addressed this below. The updates should now
* No grouping by default.
* Allow grouping using a character vector of keys, as many as desired.